src/overlays/actors/ovl_En_Syateki_Okuta/z_en_syateki_okuta.c

```diff
@@ -121,187 +120,241 @@ void EnSyatekiOkuta_Destroy(Actor* thisx, PlayState* play) {
     EnSyatekiMan* syatekiMan = (EnSyatekiMan*)this->actor.parent;
-
-    temp_v1 = EN_SYATEKI_OKUTA_GET_F(&this->actor);
-    if ((temp_v1 == 1) || (temp_v1 == 4)) {
-        temp_a0 = syatekiMan->octorokFlags;
-        temp_a1 = (temp_v1 * 2) + 6;
-
-        if ((temp_a0 >> temp_a1) & 3) {
+    s32 index = SG_OCTO_GET_INDEX(&this->actor);
+
+    // Only the Octoroks in the center column can be obscured by another Octorok. The Octoroks in the
+    // left and right columns are always visible, since the player looks at them from an angle.
+    // Additionally, the Octorok in the front row is always visible, even if it's in the center column.
+    if ((index == SG_OCTO_INDEX_FOR_POS(SG_OCTO_ROW_BACK, SG_OCTO_COL_CENTER)) ||
+        (index == SG_OCTO_INDEX_FOR_POS(SG_OCTO_ROW_CENTER, SG_OCTO_COL_CENTER))) {
+        // Checks to see if this Octorok is hidden by an Octorok immediately in front of it.
+        if (SG_OCTO_GET_TYPE(syatekiMan->octorokFlags, SG_OCTO_INDEX_DIRECTLY_IN_FRONT(index)) != SG_OCTO_TYPE_NONE) {
             return true;
         }
 
-        if ((temp_a1 == 8) && ((temp_a0 >> 0xE) & 3)) {
+        // If this Octorok is in the back row, it can also be hidden by an Octorok in the front row.
+        if ((index == SG_OCTO_INDEX_FOR_POS(SG_OCTO_ROW_BACK, SG_OCTO_COL_CENTER)) &&
+            (SG_OCTO_GET_TYPE(syatekiMan->octorokFlags, SG_OCTO_INDEX_FOR_POS(SG_OCTO_ROW_FRONT, SG_OCTO_COL_CENTER)) !=
+             SG_OCTO_TYPE_NONE)) {
             return true;
         }
     }
 
     return false;
 }
 
-void func_80A36260(EnSyatekiOkuta* this) {
+void EnSyatekiOkuta_SetupAttachToShootingGalleryMan(EnSyatekiOkuta* this) {
     Animation_PlayOnceSetSpeed(&this->skelAnime, &gOctorokAppearAnim, 0.0f);
     this->actor.draw = NULL;
-    this->actionFunc = func_80A362A8;
+    this->actionFunc = EnSyatekiOkuta_AttachToShootingGalleryMan;
 }
 
-void func_80A362A8(EnSyatekiOkuta* this, PlayState* play) {
+/**
+ * Checks every NPC actor in the scene to find the Shooting Gallery Man. Once it finds him, this will
+ * make him the parent to the Octorok. This is required because the Octoroks are normally spawned as
+ * part of the Town Shooting Gallery scene, so they don't have anything that links them to the Shooting
+ * Gallery Man, and the Octoroks need a pointer to him in order to access his Octorok flags. If this
+ * actor is spawned in a scene *without* the Shooting Gallery Man, its action function will never change
+ * from this function, and the Octorok will effectively do nothing.
+ */
+void EnSyatekiOkuta_AttachToShootingGalleryMan(EnSyatekiOkuta* this, PlayState* play) {
     Actor* actorIt = play->actorCtx.actorLists[ACTORCAT_NPC].first;
 
     while (actorIt != NULL) {
         if (actorIt->id == ACTOR_EN_SYATEKI_MAN) {
             this->actor.parent = actorIt;
-            func_80A362F8(this);
+            EnSyatekiOkuta_SetupDoNothing(this);
             break;
         } else {
             actorIt = actorIt->next;
         }
```

Comment:
```suggestion
            EnSyatekiOkuta_SetupDoNothing(this);
            break;
        }
        actorIt = actorIt->next;
```
Just personal preference